src/assets/CurrencyRateController.ts

```diff
@@ -157,16 +174,21 @@ export class CurrencyRateController extends BaseController<CurrencyRateConfig, C
     }
     const releaseLock = await this.mutex.acquire();
     try {
-      const { conversionDate, conversionRate } = await this.fetchExchangeRate(
+      const { conversionDate, conversionRate, usdConversionRate } = await this.fetchExchangeRate(
         this.activeCurrency,
         this.activeNativeCurrency,
+        this.includeUSDRate,
       );
-      this.update({
+      const newState: CurrencyRateState = {
         conversionDate,
         conversionRate,
         currentCurrency: this.activeCurrency,
         nativeCurrency: this.activeNativeCurrency,
-      });
+      };
+      if (this.includeUSDRate) {
+        newState.usdConversionRate = usdConversionRate;
+      }
```

Comment:
Similarly, we should probably re-assign `usdConversionRate` in all cases here, so that a stale value doesn't remain in state indefinitely in the case where `includeUSDRate` is disabled at runtime.

```suggestion
        usdConversionRate: this.includeUSDRate ? usdConversionRate : this.defaultState.usdConversionRate,
      };
```